### PR TITLE
Implement round joining functionality

### PR DESCRIPTION
**Pull request by sergioabplanalp**

### CODE REVIEW

File: player/types.go [hunk 1]
```diff
@@ -36,9 +36,9 @@ func (PlayerRoundStatus) ShiftStatus() {
 
 var (
 	PlayerRoundStatusUnknown         PlayerRoundStatus = 0
+	PlayerRoundStatusRoundJoining    PlayerRoundStatus = 1
```
Comment (tom-knoetze) [inline player/types.go:39]: thanks :D

File: player/ops/b_joining.go [hunk 2]
```diff
@@ -1,4 +1,63 @@
...
 
+import (
+	"context"
+	"github.com/adamhicks/ctrlaltdefeat/player"
+	"github.com/adamhicks/ctrlaltdefeat/player/config"
+	"github.com/adamhicks/ctrlaltdefeat/player/db/cursors"
+	"github.com/adamhicks/ctrlaltdefeat/player/db/rounds"
+	"github.com/corverroos/unsure"
+	"github.com/corverroos/unsure/engine"
+	_ "github.com/corverroos/unsure/engine"
```
Comment (wesrobin-luno) [inline player/ops/b_joining.go]: Remove

File: player/ops/b_joining.go [hunk 3]
```diff
@@ -1,4 +1,63 @@
...
+
+const roundJoiningConsumer = "round_joining"
+
+func JoinRoundsForever(p config.Config, b Backends) {
+	f := func(ctx context.Context, fate fate.Fate, e *reflex.Event) error {
+		if !reflex.IsType(e.Type, engine.EventTypeRoundJoin) {
+			return fate.Tempt()
+		}
+
+		pr, err := rounds.LookupRoundAndStatus(ctx, b.DB(), e.ForeignIDInt(), player.PlayerRoundStatusRoundJoining)
```
Comment (tom-knoetze) [inline player/ops/b_joining.go:26]: @adamhicks is e.ForeignIDInt() == roundID?

File: player/ops/b_joining.go [hunk 4]
```diff
@@ -1,4 +1,63 @@
...
+		if err != nil {
+			return fate.Tempt()
```
Comment (wesrobin-luno) [inline player/ops/b_joining.go]: err

File: player/ops/b_joining.go [hunk 5]
```diff
@@ -1,4 +1,63 @@
...
+			return err
+		}
+
+		tx, err := b.DB().Begin()
+		if err != nil {
+			return err
+		}
+
+		if joined {
+			_, err := rounds.Joined(ctx, tx, pr.RoundID)
```
Comment (adamhicks) [inline player/ops/b_joining.go]: this takes pr.ID not pr.RoundID

File: player/ops/b_joining.go [hunk 6]
```diff
@@ -1,4 +1,63 @@
...
+			if err != nil {
+				return err
+			}
+
+			return fate.Tempt()
+		}
+
+		_, err = rounds.Excluded(ctx, tx, pr.RoundID)
```
Comment (adamhicks) [inline player/ops/b_joining.go]: same

**Pull request merged**